Sources/Environment/Environment.swift

```diff
@@ -44,17 +44,18 @@ public struct Environment {
     public func all() -> [String: String] {
         var env = environ
         var pairs: [String: String] = [:]
-        while true {
-            guard let cval = env?.pointee else { break }
-            let pairString = String(cString: cval)
+        while let cpair = env?.pointee {
+            defer { env = env?.successor() }
+
+            let pairString = String(cString: cpair)
             let pair = pairString
-                .characters
+                .unicodeScalars
```

Comment:
I'm not sure why you'd want to use `unicodeScalars` here. The huge advantage of `characters` is that Swift already handles all the unicode handling of composed characters etc, so it should be safer. 

What's the performance difference? In which cases/inputs does it produce a different output?